cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -228,6 +230,17 @@ func ReadinessProbe(host, path string, port int, scheme v1.URIScheme) *v1.Probe
 	return createHTTPProbe(host, path, port, scheme, 0, 15, 3, 1)
 }
 
+// StartupProbe creates a Probe object with a HTTPGet handler
+func StartupProbe(host, path string, port int, scheme v1.URIScheme, timeoutForControlPlane *metav1.Duration) *v1.Probe {
+	periodSeconds, totalDuration := int32(10), constants.DefaultControlPlaneTimeout.Seconds()
+	if timeoutForControlPlane != nil {
+		totalDuration = timeoutForControlPlane.Seconds()
+	}
+	// dynamically set the value of `failureThreshold` to meet the maximum timeout for ControlPlane static Pods
+	failureThreshold := int32(math.Ceil(totalDuration / float64(periodSeconds)))
```

Comment:
Shouldn't this rather be:

```
failureThreshold := int32(math.Ceil(float64(totalDuration - 15) / float64(periodSeconds)))
```

to take the `initialDelaySeconds` into account?

Reply:
Thanks @johscheuer 
But I think it's more robust over a longer startup timeout duration, right?

Reply:
Since `kubeadm` will probably fail when the timeout is reached I don't see how this makes the setup more robust. But for simplicity your method should be good (maybe add a sentence to the comment that we ignore the `initialDelaySeconds` for simplicity (otherwise someone will try yo fix it).

Reply:
we discussed briefly this PR in the kubeadm office hours.
the existing failureThreshold calculation seems overall fine, but @rosti will provide extra feedback if needed.

Reply:
> Since `kubeadm` will probably fail when the timeout is reached I don't see how this makes the setup more robust. But for simplicity your method should be good (maybe add a sentence to the comment that we ignore the `initialDelaySeconds` for simplicity (otherwise someone will try yo fix it).

Yes, maybe we should add some comments about this.